### DISCUSSION TITLE
pre-split mesh periodic condition in a multiapp

### DISCUSSION THREAD

**Discussion by KhaledNabilSharafeldin** [idaholab/moose#21738] [Q&A General] [answered]
Hello,
Currently I have a 3 nested multiapps let's call them (A, B, C), where A hands over to B hands over to C, 
I am going to focus on the first two (A & B). The main app (A) and subapp (B) both have manual periodic BC with translation vectors along x, y, z.
I create the sidesets for left, right, etc, and use `clone_parent_mesh = true`.

The periodic BC on main app A runs without issues, while the one in subapp B results in `periodic boundary neighbor not found` error. when the BC in subapp B is commented out, it starts up normally.

I also tried running subapp B and C without A, and it seems to be working without removing the periodic BC.

is there a specific reason or fix for this?

Best,
Khaled

**Comment by GiudGiud** [comment, marked as answer]
Seems like a bug. Can you please make a minimal (non)working example for this?

The temporary fixup should be very easy, just dont clone the master mesh, load it twice
